api/resolvers/wallet.js

```diff
@@ -47,14 +47,19 @@ function injectResolvers (resolvers) {
         })
       }
 
-      const validData = await validateWallet(walletDef,
-        { ...data, ...settings, vaultEntries: vaultEntries ?? existingVaultEntries },
-        { serverSide: true })
-      if (validData) {
-        data && Object.keys(validData).filter(key => key in data).forEach(key => { data[key] = validData[key] })
-        settings && Object.keys(validData).filter(key => key in settings).forEach(key => { settings[key] = validData[key] })
+      const validate = async ({ data, settings, skipGenerated = false }) => {
+        const validData = await validateWallet(walletDef,
+          { ...data, ...settings, vaultEntries: vaultEntries ?? existingVaultEntries },
+          { serverSide: true, skipGenerated })
+        if (validData) {
+          data && Object.keys(validData).filter(key => key in data).forEach(key => { data[key] = validData[key] })
+          settings && Object.keys(validData).filter(key => key in settings).forEach(key => { settings[key] = validData[key] })
+        }
```

Comment:
this becomes a function because we need to run it twice.
- Before testCreateInvoice with skipGenerated=true (because we don't have the generated properties yet)
- After testCreateInvoice with skipGenerated=false